js/bundle/src/cli.ts

```diff
@@ -37,55 +38,82 @@ export function addFilesRecursively(
   }
 }
 
-export function main() {
-  const options = commander
+function readOptions() {
+  return commander
     .requiredOption('-d, --dir <directory>', 'input root directory (required)')
     .option('-b, --baseURL <URL>', 'base URL')
     .option(
       '-f, --formatVersion <formatVersion>',
-      'webbundle format version, possible values are "b1" and "b2" (default: "b2")',
-      'b2'
+      'webbundle format version, possible values are ' +
+        APPROVED_VERSIONS.map((v) => `"${v}"`).join(' and ') +
+        ' (default: "' +
+        DEFAULT_VERSION +
+        '")',
+      DEFAULT_VERSION
     )
     .option(
       '-p, --primaryURL <URL>',
-      'primary URL (defaults to base URL, only valid with format version "b1")'
+      'primary URL (defaults to base URL, only valid with format version "' +
+        B1 +
+        '")'
     )
     .option(
       '-m, --manifestURL <URL>',
-      'manifest URL (only valid with format version "b1")'
+      'manifest URL (only valid with format version "' + B1 + '")'
     )
     .option('-o, --output <file>', 'webbundle output file', 'out.wbn')
     .parse(process.argv);
+}
 
+function optionsValid(options: any) {
   if (options.baseURL && !options.baseURL.endsWith('/')) {
     console.error("error: baseURL must end with '/'.");
     process.exit(1);
   }
 
-  if (options.formatVersion === undefined || options.formatVersion === 'b2') {
-    // webbundle format version b2
-    const builder = new BundleBuilder();
-    if (options.primaryURL) {
-      builder.setPrimaryURL(options.primaryURL);
-    }
-    addFilesRecursively(builder, options.baseURL || '', options.dir);
-    fs.writeFileSync(options.output, builder.createBundle());
-  } else if (options.formatVersion === 'b1') {
-    // webbundle format version b1
-    const primaryURL = options.primaryURL || options.baseURL;
-    if (!primaryURL) {
-      console.error('error: Primary URL is required.');
-      process.exit(1);
-    }
-    const builder = new BundleBuilder('b1');
-    builder.setPrimaryURL(primaryURL);
+  if (
+    options.formatVersion !== undefined &&
+    !APPROVED_VERSIONS.includes(options.formatVersion)
+  ) {
+    console.error(
+      'error: invalid format version (must be ' +
+        APPROVED_VERSIONS.map((val) => `"${val}"`).join(' or ') +
+        ').'
+    );
+    process.exit(1);
+  }
+
+  if (
+    options.formatVersion === B1 &&
+    !(options.primaryURL || options.baseURL)
+  ) {
+    console.error('error: Primary URL is required.');
+    process.exit(1);
+  }
+}
+
+export function main() {
+  const options = readOptions();
+  optionsValid(options);
+
+  const version =
+    options.formatVersion === undefined
+      ? DEFAULT_VERSION
+      : options.formatVersion;
```

Comment:
If support for Node.js 12 is dropped at some point, then this could be rewritten as `const version = options.formatVersion ?? DEFAULT_VERSION` (`??` was introduced in Node.js 14).

Reply:
Yeah, since that's not the case yet, I didnt use that yet

Reply:
Just a note; since this file is transpiled to JS, we should be able to use new syntaxes as long as we set appropriate `target` in `tsconfig.json`.